Rdmp.Dicom/Cache/Pipeline/Ordering/HierarchyBasedOrder.cs

```diff
@@ -22,7 +23,7 @@ class HierarchyBasedOrder : IOrder
         private readonly DateTime _dateTo;
         private readonly HierarchyBasedOrder parent;
         private Queue<HierarchyBasedPicker> _pickers;
-        private readonly HashSet<Item> retried = new HashSet<Item>();
+        private readonly ConcurrentDictionary<Item,int> _retried = new ConcurrentDictionary<Item,int>();
```

Comment:
Do we want this to be static? would make sense to maintain a application lifetime count of how often we have tried to get it?

Reply:
Probably not AIUI - we should have a single HierarchyBasedOrder instance for each target hour, and each study's timestamp will only be in one anyway - so that would only be helpful if we ended up re-fetching the same hour during one run, in which case we might be wanting a fresh start anyway?